platform/linux-generic/odp_pool.c

```diff
@@ -232,19 +237,40 @@ static void init_buffers(pool_t *pool)
 	uint32_t mask;
 	int type;
 	uint32_t seg_size;
+	uint64_t page_size = 0;
+	int skipped_blocks = 0;
+
+	if (odp_shm_info(pool->shm, &shm_info) == 0)
+		page_size = shm_info.page_size;
```

Comment:
if call fails it has to be an error? result in that case is not predictable?

Reply:
@muvarov In the unlikely (impossible in the case of the current linux-generic implementation) event of this call failing, `page_size` will be left at the initialized default 0 value, which means we won't skip any buffers since 0 is less than `FIRST_HP_SIZE`. No no error here.

Reply:
As Bill noted, if odp_shm_info() could somehow fail, nothing is still broken done.

Reply:
actually it can.  it can return -1 for some reasons. And pool_create() and pool_destroy() from different threads for the same time, will make shm invalid before init_buffers(). But that is corner case and programmable error :)

But it works. My point was a little bit different. In my understanding assuming that odp api function will fail is not correct in places like that. That works like work around - what we will do if call will fail. Instead of that you should force platform to implement this api call in correct way. Especially if it's reference implementation. I would prefer ODP_ABORT() if call fails as that is not expected here.

Reply:
This is happening within `odp_pool_create()`, which is creating the `odp_pool_t`, so other threads don't have that handle with which to call `odp_pool_destroy()`. 

The only way `odp_shm_info()` can fail is if it's passed an unknown `odp_shm_t` handle, which is not the case here. 

I have no objection to adding an `else ODP_ABORT()` here, however.

Reply:
OK, I'll add an abort branch.

Reply:
@Bill-Fischofer-Linaro  odp_pool_lookup() works in any thread. shm look up also can get that memory. Destroy path also has look up and destroy. So I think simple ABORT() here is good.